src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -550,7 +570,9 @@ public void addChild(FieldFrame child) {
          * @param field the ast of the field.
          */
         public void addField(DetailAST field) {
-            fields.add(field);
+            if (field.findFirstToken(TokenTypes.IDENT) != null) {
```

Comment:
This is to prevent some NPEs where parameters had no `IDENT` but `this`.
See test cases for example.